osu.Game/Screens/Play/BreaksOverlay/BreakOverlay.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Rulesets.Scoring;
+using System.Collections.Generic;
+
+namespace osu.Game.Screens.Play.BreaksOverlay
+{
+    public class BreakOverlay : Container
+    {
+        private const double fade_duration = BreakPeriod.MIN_BREAK_DURATION / 2;
+        private const float remaining_time_container_max_size = 0.3f;
+        private const int vertical_margin = 25;
+
+        public List<BreakPeriod> Breaks;
+
+        private readonly bool letterboxing;
+        private readonly LetterboxOverlay letterboxOverlay;
+        private readonly Container remainingTimeBox;
+        private readonly RemainingTimeCounter remainingTimeCounter;
+        private readonly InfoContainer info;
+        private readonly ArrowsOverlay arrowsOverlay;
+
+        public BreakOverlay(bool letterboxing)
+        {
+            this.letterboxing = letterboxing;
+
+            RelativeSizeAxes = Axes.Both;
+            Children = new Drawable[]
+            {
+                letterboxOverlay = new LetterboxOverlay
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                },
+                remainingTimeBox = new Container
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.X,
+                    Size = new Vector2(0, 8),
+                    CornerRadius = 4,
+                    Masking = true,
+                    Child = new Box { RelativeSizeAxes = Axes.Both }
+                },
+                remainingTimeCounter = new RemainingTimeCounter
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.BottomCentre,
+                    Margin = new MarginPadding { Bottom = vertical_margin },
+                },
+                info = new InfoContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.TopCentre,
+                    Margin = new MarginPadding { Top = vertical_margin },
+                },
+                arrowsOverlay = new ArrowsOverlay
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            InitializeBreaks();
+        }
+
+        public void InitializeBreaks()

```

Comment:
This shouldn't be public, and should be automatically called on setting `Breaks`. Make sure when you make this change to clear all existing transforms (including scheduled ones), as there's currently an easily reproducible bug using your test cases (run the two-breaks testcase then any other; they will overlap and cause unexpected results).